packages/transitive-overlay/src/route-layers.ts

```diff
@@ -60,11 +50,33 @@ export function patternToRouteFeature(
  */
 export function getRouteLayerLayout(textField: string): SymbolLayout {
   return {
+    // "icon-image": "debug",
+    "icon-image": [
+      "case",
+      ["==", ["length", ["get", textField]], 1],
+      "circle-1",
+      ["==", ["length", ["get", textField]], 2],
+      "circle-2",
+      ["==", ["length", ["get", textField]], 3],
+      "circle-3",
+      "rect"
```

Comment:
I think the default behavior should be a rectangle so it is like the current behavior. If circles/pills are needed, how about adding a prop that takes a lookup, falling back on the rectangle if none:
```
// length-based lookup
{
  1: "circle-1",
  2: "circle-2",
  // etc
  // then convert into style-specification syntax
}
```